src/link.rs

```diff
@@ -27,101 +31,129 @@ impl Link {
     pub fn new(url: Url, title: String, desc: Option<String>) -> Self {
         Self {
             url,
-            meta: Metadata { title, desc },
+            prop: Properties { title, desc },
         }
     }
 
     pub fn id(&self) -> Result<ResourceId> {
         ResourceId::compute_bytes(self.url.as_str().as_bytes())
     }
 
-    /// Load a link with its metadata from file
+    fn load_user_data<P: AsRef<Path>>(
+        root: P,
+        id: &ResourceId,
+    ) -> Result<Properties> {
+        let path = root
+            .as_ref()
+            .join(ARK_FOLDER)
+            .join(PROPERTIES_STORAGE_FOLDER)
+            .join(id.to_string());
+        let file = AtomicFile::new(path)?;
+        let current = file.load()?;
+        let data = current.read_to_string()?;
+        let user_meta: Properties = serde_json::from_str(&data)?;
```

Comment:
Am I right, `data` can be array here?

Reply:
The struct Can have properties as Vector. 
Is it an issue if I update the definition of Properties?